paymentsheet/src/main/java/com/stripe/android/link/LinkActivityViewModel.kt

```diff
@@ -232,21 +226,38 @@ internal class LinkActivityViewModel @Inject constructor(
     }
 
     /**
-     * Attempts to confirm payment eagerly if the Link flow was launched with an authenticated user and a preselected
-     * Link payment method.
-     * @return The result of the confirmation attempt, or null if no confirmation was attempted.
+     * Attempts to confirm payment eagerly if the Link flow was launched in confirmation mode.
      */
-    private suspend fun confirmPreselectedLinkPaymentIfAvailable(): LinkConfirmationResult? {
-        val selectedPayment = (linkLaunchMode as? LinkLaunchMode.Full)?.selectedPayment
-        val paymentReadyForConfirmation = selectedPayment?.takeIf { it.readyForConfirmation() }
-        val account = linkAccount ?: return null
-        return paymentReadyForConfirmation?.let {
-            linkConfirmationHandler.confirm(
-                paymentDetails = it.details,
-                cvc = it.collectedCvc,
-                linkAccount = account,
+    private suspend fun confirmLinkPayment(selectedPayment: LinkPaymentMethod) = runCatching {
+        require(selectedPayment.readyForConfirmation()) { "LinkPaymentMethod must be ready for confirmation" }
+        linkConfirmationHandler.confirm(
+            paymentDetails = selectedPayment.details,
+            cvc = selectedPayment.collectedCvc,
+            linkAccount = requireNotNull(linkAccount) { "LinkAccount must not be null for confirmation" },
+        )
+    }.onSuccess { confirmResult ->
+        dismissWithResult(
+            when (confirmResult) {
+                LinkConfirmationResult.Canceled -> LinkActivityResult.Canceled(
+                    reason = LinkActivityResult.Canceled.Reason.BackPressed,
+                    linkAccountUpdate = LinkAccountUpdate.None
+                )
+                is LinkConfirmationResult.Failed -> LinkActivityResult.Failed(
+                    IllegalStateException("Failed to confirm Link payment: ${confirmResult.message}"),
+                    linkAccountUpdate = LinkAccountUpdate.None
+                )
+                LinkConfirmationResult.Succeeded -> LinkActivityResult.Completed(
+                    linkAccountUpdate = LinkAccountUpdate.Value(null)
+                )
+            }
+        )
+    }.onFailure {
+        dismissWithResult(
+            LinkActivityResult.Failed(
+                IllegalStateException("Failed to confirm Link payment: "),
+                linkAccountUpdate = LinkAccountUpdate.None
             )
-        }
+        )
```

Comment:
when launching for Confirmation only, ensure no UI is presented and confirmation result gets populated to FlowController.